src/main/java/com/timgroup/statsd/NonBlockingStatsDClient.java

```diff
@@ -469,6 +445,17 @@ StringBuilder tagString(final String[] tags, StringBuilder builder) {
         return tagString(tags, constantTagsRendered, builder);
     }
 
+    ClientChannel createByteChannel(Callable<SocketAddress> addressLookup, int timeout, int bufferSize) throws Exception {
+        final SocketAddress address = addressLookup.call();
+        if (address instanceof NamedPipeSocketAddress) {
+            return new NamedPipeClientChannel((NamedPipeSocketAddress) address);
+        } else if (address instanceof UnixSocketAddress) {
+            return new UnixDatagramClientChannel(address, timeout, bufferSize);
+        } else {
+            return new DatagramClientChannel(DatagramChannel.open(), address);
```

Comment:
Would this work?
```suggestion
            return new DatagramClientChannel(address);
```
And then DatagramChannel import can be removed, and maybe 2-argument constructor too (other client channel impls don't have one, so I guess we don't need one there too).